# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/basetypes/Accessible.cpp
# Commit: 5d76535b47fc
# Full Hash: 5d76535b47fced08e46753a0262211732fb26417
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2022-01-05 09:33:53
# Regressor Bug: 1737944
# File Overlap Count: 2
# Description:
#   Bug 1737944 - P7: Move and rename GetPositionAndSizeInternal to Accessible. r=Jamie
#   
#   Now that the AccGroupInfo getter is in Accessible, the position and size
#   getter could go there too.
#   
# ==============================================================================

diff -r fca1ff7eabf4 -r 5d76535b47fc accessible/basetypes/Accessible.cpp
--- a/accessible/basetypes/Accessible.cpp	Tue Jan 04 21:01:36 2022 +0000
+++ b/accessible/basetypes/Accessible.cpp	Tue Jan 04 21:01:37 2022 +0000
@@ -4,6 +4,7 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #include "Accessible.h"
+#include "AccGroupInfo.h"
 #include "ARIAMap.h"
 #include "States.h"
 #include "mozilla/a11y/HyperTextAccessibleBase.h"
@@ -210,3 +211,11 @@
 
   return level;
 }
+
+void Accessible::GetPositionAndSetSize(int32_t* aPosInSet, int32_t* aSetSize) {
+  auto groupInfo = GetOrCreateGroupInfo();
+  if (groupInfo) {
+    *aPosInSet = groupInfo->PosInSet();
+    *aSetSize = groupInfo->SetSize();
+  }
+}